src/cli/tls_helpers.h

```diff
@@ -163,6 +161,12 @@ class Basic_Credentials_Manager : public Botan::Credentials_Manager {
                   continue;
                }
 
+               if(std::find(cert_signature_schemes.begin(),
+                            cert_signature_schemes.end(),
+                            i.certs[0].subject_public_key_algo()) == cert_signature_schemes.end()) {
+                  continue;
+               }
```

Comment:
This is the reason that the ASIO tests fail to perform a successful handshake. The algorithm identifiers of the signature scheme involve the padding (e.g. "RSA/PKCS1v15(SHA-256)") but the subject public key's algorithm identifier is just that, e.g. "RSA", without the padding.

The ASIO integration tests happen to use an RSA certificate that doesn't pass this check. So the server fails to find a suitable certificate and raises a handshake failure.